### ISSUE TITLE
VB.NET - Incorrect error BC30506 when compiling with .NET 5.0

### ISSUE THREAD

**Issue by ehasis** [dotnet/roslyn#49276]
Migrating a solution from .NET Core 3.1 to .NET 5.0 fail to compile. I was able to reproduce the problem with a minimal project as described on the steps bellow.

**Version Used**: 16.8.0

**Steps to Reproduce**:
1. Create a VB .NET Core Console Application (ConsoleApp1)
2. Add a new VB .NET Library project (Core) (ClassLibrary1)
3. Change TFM to .NET 5.0 on both projects
4. Make the ConsoleApp1 project references the ClassLibrary1
5. Add the following classes on ClassLibrary1
```vb
Public Class Button
    Public Event Click As EventHandler
End Class

Public Class Base
    Protected WithEvents Button1 As Button
End Class
```
6. Add the following class on ConsoleApp1
```vb
Public Class Derived
    Inherits ClassLibrary1.Base
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
    End Sub
End Class
```

**Expected Behavior**:
The solution successfully compile.

**Actual Behavior**:
Fail compile the line `Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click` with:
error BC30506: Handles clause requires a WithEvents variable defined in the containing type or one of its base types.

**Comment by ProDInfo** [comment]
I have the same problem with a WPF project with the lastest Visual studio 2019 (16.8)
When compiling for net framework the same error BC30506

**Comment by jaredpar** [comment]
@AlekseyTs for analysis.

**Comment by AlekseyTs** [comment]
It looks like the ref assembly produced by the build for the library is lacking definition for the field backing WithEvents property:
```
.field private class ClassLibrary4.Button _Button1
.custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
.custom instance void [System.Runtime]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 00 00 00 00 00 00 ) 
.custom instance void [System.Runtime]System.Runtime.CompilerServices.AccessedThroughPropertyAttribute::.ctor(string) = ( 01 00 07 42 75 74 74 6F 6E 31 00 00 )             // ...Button1..
```

The original library has it and the build against it succeeds. The presence of that field with combination of AccessedThroughPropertyAttribute on it pointing to the property is required for WithEvents feature to work across assembly boundaries. I am not sure what tool is used to produce the ref assembly, but it looks like it should be adjusted to keep fields like that around.

**Comment by AlekseyTs** [comment]
@jaredpar 
> for analysis.

Done.

**Comment by jaredpar** [comment]
Thanks @AlekseyTs, 

Moving to the winforms repo as I believe they own the ref assemblies for these controls.
